Fix EdDSA (signing is missing, signature packet doesn't parse)

zapu opened the pull request:
Just getting started, some non-passing tests for now.

zapu committed:
Add tests for EdDSA

zapu committed:
Move ed25519 to go-crypto

zapu committed:
Add signing for EdDSA private keys (ed25519)

zapu commented:
Working now, but I want to move eddsa stuff to separate package or at least separate file.

zapu commented on openpgp/packet/public_key.go:
```diff
@@ -65,18 +65,17 @@ type edDSAkey struct {
 	// reading and keep only actual number in p field. Find out how
 	// other MPIs are stored.
-	copy(key[:], e.p.bytes[1:])
+	key := e.p.bytes[1:]
```
we should check here that we `e.p.bytes` is right size, because `ed25519` will just panic if it gets wrong sized buffers IIRC.

maxtaco replied:
This seems especially bad since it could be running on untrusted input.  Can you add a check?

zapu committed:
Add more EdDSA/cv25519 tests

zapu committed:
EdDSAPrivateKey.Sign style fix

zapu committed:
Add EDDSA test key that panics

zapu committed:
Add checks for EdDSA key lengths

Also refactored invalid eddsa key test, now there are two malformed
keys, one with bad private key, one with bad public key.
